map/elevation_info.cpp

```diff
@@ -2,70 +2,54 @@
 
 #include "geometry/mercator.hpp"
 
-#include "base/string_utils.hpp"
-
-namespace
-{
-static uint8_t constexpr kMaxDifficulty = ElevationInfo::Difficulty::Hard;
-
-std::string const kAscentKey = "ascent";
-std::string const kDescentKey = "descent";
-std::string const kLowestPointKey = "lowest_point";
-std::string const kHighestPointKey = "highest_point";
-std::string const kDifficultyKey = "difficulty";
-std::string const kDurationKey = "duration";
-
-template <typename T>
-void FillProperty(kml::Properties const & properties, std::string const & key, T & value)
-{
-  auto const it = properties.find(key);
-  if (it == properties.cend())
-    LOG(LERROR, ("Property not found for key:", key));
-  else
-  {
-    if (!strings::to_any(it->second, value))
-      LOG(LERROR, ("Conversion is not possible for key", key, "string representation is", it->second));
-  }
-}
-}  // namespace
-
-ElevationInfo::ElevationInfo(Track const & track)
-  : m_id(track.GetId())
-  , m_name(track.GetName())
+ElevationInfo::ElevationInfo(kml::MultiGeometry const & geometry)
 {
-  // (Distance, Elevation) chart doesn't have a sence for multiple track's geometry.
-  auto const & points = track.GetSingleGeometry();
-  if (points.empty())
-    return;
+  double distance = 0;
 
-  m_points.reserve(points.size());
-  m_points.emplace_back(0, points[0].GetAltitude());
-  double distance = 0.0;
-  for (size_t i = 1; i < points.size(); ++i)
+  // Concatenate all segments.
+  for (size_t i = 0; i < geometry.m_lines.size(); ++i)
   {
-    distance += mercator::DistanceOnEarth(points[i - 1].GetPoint(), points[i].GetPoint());
-    m_points.emplace_back(distance, points[i].GetAltitude());
+    auto const & points = geometry.m_lines[i];
+    if (points.empty())
+      continue;
+
+    double distanceToSegmentBegin = 0;
+    if (i == 0)
+    {
+      m_minAltitude = points[i].GetAltitude();
+      m_maxAltitude = m_minAltitude;
+    }
+    else
+    {
+      distanceToSegmentBegin = m_points.back().m_distance;
+      m_segmentsDistances.emplace_back(distanceToSegmentBegin);
+    }
+
+    m_points.emplace_back(points[i], distanceToSegmentBegin);
```

Comment:
Tej elevation profile is the data only for drawing the chart.

IF the track has multiple segments the may hot have common points. 
Like you record one part, pause the recording and then after some hours you continue the recording from the other place - it will be multisegment trac recording.

And there are several approach watches how to display such kind of track:
1. We can show the profile for the each segment, but it may overload the UI or we should have some switcher between segments
2. Show the gap between end of 1segm and start of another. But it also may looks ugly or strange
3. Join the profiles into the one line. In this case the start of next segment will be equal to the end of the previous but with another altitude to properly draw the chart line.

There is no reason to store the points in the same way as they stored in the track. The elevation profile points should be stored in a way as the profile should be drawn on the screen.

Reply:
Fixed!